driver/razerkbd_driver.c

```diff
@@ -552,6 +552,69 @@ static ssize_t razer_attr_read_game_led_state(struct device *dev, struct device_
     return sprintf(buf, "%d\n", response.arguments[2]);
 }
 
+/**
+ * Write device file "keyswitch_optimization"
+ */
+static ssize_t razer_attr_write_keyswitch_optimization(struct device *dev, struct device_attribute *attr, const char *buf, size_t count)
+{
+    struct usb_interface *intf = to_usb_interface(dev->parent);
+    struct usb_device *usb_dev = interface_to_usbdev(intf);
+    struct razer_report report = {0};
+    unsigned char mode = (unsigned char)simple_strtoul(buf, NULL, 10);
+
+    // Toggle Keyswitch Optimization
+    switch(usb_dev->descriptor.idProduct) {
+    case USB_DEVICE_ID_RAZER_HUNTSMAN_V2:
+        report = razer_chroma_misc_set_keyswitch_optimization(NOSTORE, mode);
+        report.transaction_id.id = 0x1f;
+        razer_send_payload(usb_dev, &report);
+        report = razer_chroma_misc_set_keyswitch_optimization(VARSTORE, mode);
+        report.transaction_id.id = 0x1f;
+        razer_send_payload(usb_dev, &report);
+        break;
+    default:
+        report = razer_chroma_misc_set_keyswitch_optimization(NOSTORE, mode);
+        report.transaction_id.id = 0x1f;
+        razer_send_payload(usb_dev, &report);
+        report = razer_chroma_misc_set_keyswitch_optimization(VARSTORE, mode);
+        report.transaction_id.id = 0x1f;
+        razer_send_payload(usb_dev, &report);
+    }
+
+    return count;
+}
+
+/**
+ * Read device file "keyswitch_optimization"
+ */
+static ssize_t razer_attr_read_keyswitch_optimization(struct device *dev, struct device_attribute *attr, char *buf)
+{
+    struct usb_interface *intf = to_usb_interface(dev->parent);
+    struct usb_device *usb_dev = interface_to_usbdev(intf);
+    struct razer_report report = {0};
+    struct razer_report response = {0};
+    int state;
+
+    switch(usb_dev->descriptor.idProduct) {
+    case USB_DEVICE_ID_RAZER_HUNTSMAN_V2:
+        report = razer_chroma_misc_get_keyswitch_optimization();
+        report.transaction_id.id = 0x1f;
+        break;
+    default:
+        report = razer_chroma_misc_get_keyswitch_optimization();
```

Comment:
Why not as in other functions set report where the variable is declared to this and remove `default:`?

Reply:
That's true, we don't have other keyboards with varying calls at this time. I had used the `default:` statement here as is done in other effect functions. But in those functions they seem to need to call different variations of functions for certain devices. [111120d](https://github.com/openrazer/openrazer/pull/1759/commits/111120d85ee13d593fd21e9adea9a0c5c45780cc)